### DISCUSSION TITLE
How to download image aka "Save image as"

### DISCUSSION THREAD

**Discussion by kyberorg** [selenide/selenide#2599] [Q&A]
I am about to exit from Russia citizenship, but embassy gives 3-4 slots per month, so I am trying to automate slot checks. 
For this I need to download captcha image. 

![mid](https://github.com/selenide/selenide/assets/4687253/2d0c6362-f4aa-43ea-a55f-a26539875395)

Image is autogenerated, so simply opening it with `open()` won't work. 

Question: how do I download this image in correct manner?

**Comment by asolntsev** [comment]
Can you please at least share the URL of this page?
Or html source code (the piece containing that image).

**Reply by kyberorg** [reply]
Yes, sure. 
https://tallinn.kdmid.ru/queue/orderinfo.aspx?id=56831&cd=a403e162&ems=2C444843

Image HTML is: 
`<img id="ctl00_MainContent_imgSecNum" src="CodeImage.aspx?id=c975" alt="Необходимо включить загрузку картинок в браузере." style="border-width:0px;width:140px;">`

**Comment by kyberorg** [comment]
I actually found working workaround:

```
$("img[id='ctl00_MainContent_imgSecNum']").getScreenshotAs(OutputType.FILE);
```

**Comment by asolntsev** [comment]
@kyberorg There is a proper solution:
```java
import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.File;


  @Test
  void canDownloadImage() throws URISyntaxException, IOException {
    open("https://tallinn.kdmid.ru/queue/orderinfo.aspx?id=56831&cd=a403e161&ems=2C444843");
    SelenideElement img = $("#ctl00_MainContent_imgSecNum").shouldBe(visible);
    String src = img.attr("src");
    File file = Selenide.download(src, 8000);
    log.info("Downloaded image: {}", file);
    BufferedImage image = ImageIO.read(file);
    assertThat(image.getWidth()).isGreaterThan(100);
    assertThat(image.getHeight()).isGreaterThan(20);
  }
```
